src/SFML/Graphics/Font.cpp

```diff
@@ -790,11 +784,16 @@ bool Font::setCurrentSize(unsigned int characterSize) const
                         sizeFound = true;
                     }
                 }
+            }
 
-                if (result == FT_Err_Invalid_Pixel_Size || !sizeFound)
-                {
-                    err() << "Failed to fallback on an available size." << std::endl;
-                }
+            // Check if we found a way to fallback to a valid size. If we couldn't, log the error.
+            if (result == FT_Err_Invalid_Pixel_Size)
+            {
+                err() << "Failed to set bitmap font size to " << characterSize << std::endl;
+                err() << "Available sizes are: ";
+                for (int i = 0; i < face->num_fixed_sizes; ++i)
+                    err() << face->available_sizes[i].width << "x" << face->available_sizes[i].height << " ";
+                err() << std::endl;
```

Comment:
In case of a scalable font, we can't say it is a "bitmap" font and there will probably be no fixed sizes to print. So we can't use the same error message for both cases (or if we do, make it less specific).